### PR TITLE
sudachi: Fix homepage, url and autoupdate

### LINKED ISSUE #1213: [Bug]: sudachi v1.0.2: download failed
### Prerequisites

- [X] I have written a descriptive issue title.
- [X] I have searched all issues/PRs to ensure it has not already been reported or fixed.
- [X] I have verified that I am using the latest version of Scoop and corresponding bucket.

### Package Name

sudachi

### Expected/Current Behaviour

Install failed due to sudachi github repo missing

### Steps to Reproduce

```console
> scoop install sudachi
Installing 'sudachi' (v1.0.2) [64bit] from 'games' bucket
Starting download with aria2 ...
Download: 08/22 18:02:07 [ERROR] CUID#8 - Download aborted. URI=https://github.com/sudachi-emu/sudachi/releases/downloadDownload: Exception: [AbstractCommand.cc:351] errorCode=3 URI=https://github.com/sudachi-emu/sudachi/releases/download/vDownload: Download Results:
Download: gid   |stat|avg speed  |path/URI
Download: ======+====+===========+=======================================================
Download: 2bfc41|ERR |       0B/s|C:/Users/[CENSORED]/scoop/cache/sudachi#v1.0.2#a9f1c6b.7z
Download: Status Legend:
Download: (ERR):error occurred.
Download: aria2 will resume download if the transfer is restarted.
Download: If there are any errors, then see the log file. See '-l' option in help/man page for details.

ERROR Download failed! (Error 3) Resource was not found
ERROR https://github.com/sudachi-emu/sudachi/releases/download/v1.0.2/v1.0.2-windows.7z
    referer=https://github.com/sudachi-emu/sudachi/releases/download/v1.0.2/
    dir=C:\Users\[CENSORED]\scoop\cache
    out=sudachi#v1.0.2#a9f1c6b.7z
```

### Possible Solution

Move the url to https://sudachi.emuplace.app/

### Scoop and Buckets Version

```console
> scoop --version
Current Scoop version:
859d1db5 (HEAD -> master, tag: v0.5.2, origin/master, origin/HEAD) chore(release): Bump to version 0.5.2 (#6080)

'main' bucket:
1ac4c9218 (HEAD -> master, origin/master, origin/HEAD) tektoncd-cli: Update to version 0.38.0

'extras' bucket:
84fd7995c (HEAD -> master, origin/master, origin/HEAD) wavebox: Update to version 10.128.3.2

'versions' bucket:
d1e628747 (HEAD -> master, origin/master, origin/HEAD) zig-dev: Update to version 0.14.0-dev.1248

'java' bucket:
84b7bb31 (HEAD -> master, origin/master, origin/HEAD) temurin11-nightly-jre: Update to version 11.0.25-3.0.ea

'games' bucket:
6b9c02115 (HEAD, upstream/master, upstream/HEAD, origin/master) veloren-weekly: Update to version 2024-08-21T17_04
```

### Scoop Config

```console
> scoop config

last_update                       scoop_repo                              scoop_branch
-----------                       ----------                              ------------
2024-08-22T18:02:06.4104021+08:00 https://github.com/ScoopInstaller/Scoop master
```

### PowerShell Version

```console
> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.19041.4780
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.19041.4780
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

### Additional Softwares

_No response_

### PR DESCRIPTION
**Pull request by limouren**
Move homepage, checkvar and autoupdate to sudachi new home https://sudachi.emuplace.app/

Closes #1213

- [x] I have read the [Contributing Guide](https://github.com/Calinou/scoop-games/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: bucket/sudachi.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
 {
-    "version": "v1.0.2",
+    "version": "v1.0.4",
```
Comment (Calinou) [inline bucket/sudachi.json:2]:
```suggestion
    "version": "1.0.4",
```

File: bucket/sudachi.json [hunk 2]
```diff
@@ -35,13 +35,18 @@
...
+            "$relPath = ((Invoke-WebRequest https://sudachi.emuplace.app$jsPath).Content | Select-String -Pattern '/releases/sudachi-windows-v[\\d.]+.7z').Matches.Value",
+            "return $relPath"
+        ],
+        "regex": "/releases/sudachi-windows-(?<version>v[\\d.]+).7z"
     },
     "autoupdate": {
         "architecture": {
             "64bit": {
-                "url": "https://github.com/sudachi-emu/sudachi/releases/download/$version/$version-windows.7z"
+                "url": "https://sudachi.emuplace.app/releases/sudachi-windows-$version.7z"
```
Comment (Calinou) [inline bucket/sudachi.json:49]:
```suggestion
                "url": "https://sudachi.emuplace.app/releases/sudachi-windows-v$version.7z"
```

File: bucket/sudachi.json [hunk 3]
```diff
@@ -35,13 +35,18 @@
...
     "checkver": {
-        "url": "https://api.github.com/repos/sudachi-emu/sudachi/releases?per_page=1",
-        "jsonpath": "$[?(@.prerelease == false)].tag_name"
+        "url": "https://sudachi.emuplace.app/",
+        "script": [
+            "$jsPath = ((Invoke-WebRequest https://sudachi.emuplace.app/).Content | Select-String -Pattern '/assets/index-\\w+.js').Matches.Value",
+            "$relPath = ((Invoke-WebRequest https://sudachi.emuplace.app$jsPath).Content | Select-String -Pattern '/releases/sudachi-windows-v[\\d.]+.7z').Matches.Value",
+            "return $relPath"
+        ],
+        "regex": "/releases/sudachi-windows-(?<version>v[\\d.]+).7z"
```
Comment (Calinou) [inline bucket/sudachi.json:44]:
```suggestion
        "regex": "/releases/sudachi-windows-v(?<version>[\\d.]+).7z"
```

**Comment by limouren** [comment]
Thanks @Calinou, that's quick! m(_ _)m

**Pull request merged**